### DISCUSSION TITLE
Add Promotion code parameter to `SetupPaymentSheetParameters`

### DISCUSSION THREAD

**Discussion by Tom3652** [flutter-stripe/flutter_stripe#1714] [Ideas]
**Is your feature request related to a problem? (please describe)**
I would like to be able to add a promotion code to a subscription. 
Currently there is no way to pass the promotion code to `Stripe.instance.initPaymentSheet`.

**Describe the solution you'd like**
I would like to have a new parameter in `SetupPaymentSheetParameters` that is called `promotionCode` that displays the equivalent of this field from the Stripe [documentation](https://docs.stripe.com/billing/subscriptions/coupons?dashboard-or-api=api) : 

<img width="340" alt="Capture d’écran 2024-03-27 à 08 35 00" src="https://github.com/flutter-stripe/flutter_stripe/assets/32631467/6febd719-fbf4-46a7-b870-4cdd66e42c4e">

**Describe alternatives you've considered**
Designing a custom UI to enter a promo code before showing Stripe modal and use it inside the backend as a separate request to check if the promotion exists.

**Additional context**
Thanks for the package !

**Comment by jonasbark** [comment]
Please upvote this ticket:
https://github.com/stripe/stripe-react-native/issues/1130

**Comment by Tom3652** [comment]
Done ! 
In the meantime i have managed to make it work with my backend

**Comment by hamzalakrati** [comment]
+1 for this feature !

**Comment by AbdullahChauhan** [comment]
+1 for this feature!
